app/controllers/participants_controller.rb

```diff
@@ -1,9 +1,15 @@
 class ParticipantsController < ApplicationController
-  
+
   before_action :set_participant, only: [:show, :edit, :update, :destroy]
 
   # GET /participants
   # GET /participants.json
+
+  def assign_santas
+    @assign_santas = Participant.all.shuffle.zip(Participant.all)
```

Comment:
If I understand the intention of this algorithm correctly, it might need some tweaking as  there is a possibility that Participants will be paired with themselves (and that person will be sad for christmas :cry: ) e.g.,
```
![Uploading serviceseeking_—__Users_markagbuya_code_serviceseeking_—_ruby_—_238×55.png…]()
```